### PR TITLE
boards: seeed: add XIAO ESP32C6 support

### PR DESCRIPTION
**Pull request by mariopaja**
This PR adds supports for XIAO ESP32C6

### CODE REVIEW

File: boards/seeed/xiao_esp32c6/doc/img/xiao_esp32c6.jpeg
Comment (nordicjm) [inline boards/seeed/xiao_esp32c6/doc/img/xiao_esp32c6.jpeg:1]: Convert images to webp ideally with transparent backgrounds then put through https://tinypng.com/

File: boards/seeed/xiao_esp32c6/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+- Cryptographic hardware acceleration: (AES-128/256, ECC, HMAC, RSA, SHA, Digital signature, Hash)
+- Random number generator (RNG)
+
+For more information, check the datasheet at `ESP32-C6 Datasheet`_ or the technical reference
+manual at `ESP32-C6 Technical Reference Manual`_.
+
+Supported Features
+==================
+
+Current Zephyr's ESP32-C6-DevKitC board supports the following features:
```
Comment (nordicjm) [inline boards/seeed/xiao_esp32c6/doc/index.rst:87]: needs board name as used for west flash with \`\` around it and call it board target

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by mariopaja** [comment]
@nordicjm It seems like twister is failing due to the sleep modes. I did also a test localy using `uart` instead of `usb_serial` and twister passed.

**Comment by nordicjm** [comment]
> @nordicjm It seems like twister is failing due to the sleep modes. I did also a test localy using `uart` instead of `usb_serial` and twister passed.

@sylvioalves

File: boards/seeed/xiao_esp32c6/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+The Zephyr SDK uses a bundled version of OpenOCD by default. You can overwrite that behavior by adding the
+``-DOPENOCD=<path/to/bin/openocd> -DOPENOCD_DEFAULT_PATH=<path/to/openocd/share/openocd/scripts>``
+parameter when building.
+
+Here is an example for building the :zephyr:code-sample:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: xiao_esp326
```
Comment (sylvioalves) [inline boards/seeed/xiao_esp32c6/doc/index.rst:212]: typo: `xiao_esp326`

File: boards/seeed/xiao_esp32c6/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+   :goals: build flash
+   :gen-args: -DOPENOCD=<path/to/bin/openocd> -DOPENOCD_DEFAULT_PATH=<path/to/openocd/share/openocd/scripts>
+
+You can debug an application in the usual way. Here is an example for the :zephyr:code-sample:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: xiao_esp326
```
Comment (sylvioalves) [inline boards/seeed/xiao_esp32c6/doc/index.rst:220]: same

**Comment by mariopaja** [comment]
> @mariopaja Hi, would you rebase and push again?

@sylvioalves done

File: boards/seeed/xiao_esp32c6/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+This board is based on the ESP32-C6 with 4MB of flash, integrating 2.4 GHz Wi-Fi 6,
+Bluetooth 5.3 (LE) and the 802.15.4 protocol. It has an USB-C port for programming
+and debugging, integrated battery charging and an U.FL external antenna connector.
+It is based on a standard XIAO 14 pin pinout.
+
+Supported Features
+==================
+
+Current Zephyr's XIAO ESP32C6 board supports the following features:
```
Comment (nordicjm) [inline boards/seeed/xiao_esp32c6/doc/index.rst:24]: Replace with board name as used by west with \`\` around it and refer to it as board target

**Review by kartben — DISMISSED** [review]

File: boards/seeed/xiao_esp32c6/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
++------------+------------+-------------------------------------+
+| LEDC       | on-chip    | pwm                                 |
++------------+------------+-------------------------------------+
+| SPI DMA    | on-chip    | spi                                 |
++------------+------------+-------------------------------------+
+| GDMA       | on-chip    | dma                                 |
++------------+------------+-------------------------------------+
+| TRNG       | on-chip    | entropy                             |
++------------+------------+-------------------------------------+
+| USB-CDC    | on-chip    | serial                              |
```
Comment (sylvioalves) [inline boards/seeed/xiao_esp32c6/doc/index.rst:49]: Would you mind adding Wi-Fi in this list?

File: boards/seeed/xiao_esp32c6/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+   :gen-args: -DOPENOCD=<path/to/bin/openocd> -DOPENOCD_DEFAULT_PATH=<path/to/openocd/share/openocd/scripts>
+
+You can debug an application in the usual way. Here is an example for the :zephyr:code-sample:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: xiao_esp32c6
+   :goals: debug
+
+
```
Comment (sylvioalves) [inline boards/seeed/xiao_esp32c6/doc/index.rst:223]: Would you remove this extra line?

**Review by marekmatej — APPROVED** [review]
LGTM

**Review by kartben — APPROVED** [review]

**Comment by mariopaja** [comment]
Hi @sylvioalves,
It seems like twister is failing due to the watchdog

**Comment by sylvioalves** [comment]
@wmrsouza

**Comment by wmrsouza** [comment]
@mariopaja
#81890 fixes the issue

**Review by sylvioalves — APPROVED** [review]

File: boards/seeed/xiao_esp32c6/xiao_esp32c6-pinctrl.dtsi [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/pinctrl/esp-pinctrl-common.h>
+#include <dt-bindings/pinctrl/esp32c6-pinctrl.h>
+#include <zephyr/dt-bindings/pinctrl/esp32c6-gpio-sigmap.h>
+
+&pinctrl {
+
+	uart0_default: uart0_default {
```
Comment (raffarost) [inline boards/seeed/xiao_esp32c6/xiao_esp32c6-pinctrl.dtsi:13]: do you want to leave `uart0` enabled and connected to these pins?
I couldn't be sure looking at the schematic, but it seemed these pins are unavailable

Comment (mariopaja) [reply boards/seeed/xiao_esp32c6/xiao_esp32c6-pinctrl.dtsi:13]: Hi,
In the schematics the used gpios are rather unclear. The best is to check the pinout file. 

Yes, I think of leaving uart enabled like the rest of the xiao_esp32xx lineup.

image

**Review by raffarost — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**